### PR TITLE
feat(ecocredit/core): add issuance date to BatchInfo

### PR DESCRIPTION
**Pull request by technicallyty**
## Description

- adds issuance_date as a field to both `EventCreateBatch` and `BatchInfo`

Closes: #680 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: proto/regen/ecocredit/v1/events.proto [hunk 1]
```diff
@@ -54,12 +54,15 @@ message EventCreateBatch {
   // quantified and verified.
   string end_date = 6;
 
+  // issuance_date is the time of the block in which the batch was issued
```
Comment (ryanchristo) [inline proto/regen/ecocredit/v1/events.proto:57]:
```suggestion
  // issuance_date is the timestamp when the credit batch was issued.
```

File: proto/regen/ecocredit/v1/state.proto [hunk 2]
```diff
@@ -132,6 +132,9 @@ message BatchInfo {
     // end_date is the end of the period during which this credit batch was
     // quantified and verified.
     google.protobuf.Timestamp end_date = 6;
+
+    // issuance_date is the block timestamp when this batch was committed to state
```
Comment (ryanchristo) [inline proto/regen/ecocredit/v1/state.proto:136]:
```suggestion
    // issuance_date is the timestamp when the credit batch was issued.
```

**Review by ryanchristo — APPROVED** [review]
Looks good to me. I think we can simplify the proto comments. Pre-approving.

File: proto/regen/ecocredit/v1/state.proto [hunk 3]
```diff
@@ -132,6 +132,9 @@ message BatchInfo {
     // end_date is the end of the period during which this credit batch was
     // quantified and verified.
     google.protobuf.Timestamp end_date = 6;
+
+    // issuance_date is the timestamp when the credit batch was issued.
+    google.protobuf.Timestamp issuance_date = 7;
```
Comment (aleem1314) [inline proto/regen/ecocredit/v1/state.proto:137]: This is a state-breaking change. Let's add an entry for this in #678.

**Review by aleem1314 — APPROVED** [review]

**Pull request merged**